### PR TITLE
Add permissions for Enterprise Search Behavioural Analytics indexes

### PR DESCRIPTION
**Pull request by tutelaris**
### Description
Currently, the Enterprise Search Search Experience team is working on the Behavioural Analytics product introduction.
This product requires adding new indexes such as:
- `.elastic-analytics-collections`
- `logs-elastic_analytics.events-{collection_name}-{date}`

This PR is dedicated to adding permissions for the indexes described above in `ELASTIC_ENTERPRISE_SEARCH_SERVER_ROLE_DESCRIPTOR`

### Related PRs/Tasks
**Task:** https://github.com/elastic/enterprise-search-team/issues/2679

### CODE REVIEW

**Review by joemcelroy — APPROVED** [review]
LGTM - the index names are correct.

File: x-pack/plugin/security/qa/service-account/src/javaRestTest/java/org/elasticsearch/xpack/security/authc/service/ServiceAccountIT.java [hunk 1]
```diff
@@ -154,11 +154,13 @@ public class ServiceAccountIT extends ESRestTestCase {
                 {
                     "names": [
                         "search-*",
+                        ".elastic-analytics-collections",
                         ".ent-search-*",
                         ".monitoring-ent-search-*",
                         "metricbeat-ent-search-*",
                         "enterprise-search-*",
                         "logs-app_search.analytics-default",
+                        "logs-elastic_analytics.events-*",
```
Comment (ywangd) [inline x-pack/plugin/security/qa/service-account/src/javaRestTest/java/org/elasticsearch/xpack/security/authc/service/ServiceAccountIT.java:163]: Given the index name is `logs-elastic_analytics.events-{collection_name}-{date}`, would it better to have it like `logs-elastic_analytics.events-*-*`? Or are there other use cases which require a slightly broader index pattern?

**Review by ywangd — COMMENTED** [review]
I just have one question that can use some clarification.

File: x-pack/plugin/security/qa/service-account/src/javaRestTest/java/org/elasticsearch/xpack/security/authc/service/ServiceAccountIT.java (hunk 1 above)
Comment (tutelaris) [reply x-pack/plugin/security/qa/service-account/src/javaRestTest/java/org/elasticsearch/xpack/security/authc/service/ServiceAccountIT.java:163]: Makes sense. Thank you!

**Comment by ywangd** [comment]
@elasticmachine update branch

**Review by ywangd — APPROVED** [review]
LGTM

**Comment by tutelaris** [comment]
@elasticmachine update branch

**Comment by tutelaris** [comment]
@elasticmachine update branch

**Pull request merged**